simulation/agent.py

```diff
@@ -227,6 +228,11 @@ def on_collide(self) -> None:
     def on_tick(self, seen_agents: List['vision.AgentView']) -> None:
         """ Agent logic goes here """
         pass
+    
+    def make_noise(self):
+        noise_event = world.NoiseEvent(Position(self.location.x - self._width/2, self.location.y - self._width/2), self)
+        self._world.add_noise(noise_event)
+        
```

Comment:
Shouldn't need the `- self._width/2` I think, `self.location` should be the centre of the agent, if not we should change it :p

Reply:
Without it looked as if the noise origins from the top right of the agent.

Reply:
Huh, that's weird

Reply:
Pretty sure that's a mistake in the rendering of the noises then, cause I'm pretty sure the collision detection wouldn't work properly otherwise.

Reply:
But it's kinda confusing, since some of the arcade drawing methods use the centre and other use the corner...

Reply:
> Pretty sure that's a mistake in the rendering of the noises then, cause I'm pretty sure the collision detection wouldn't work properly otherwise.

I think if the position of all agent/obstacles is at the top it would be the same as if they were in the center. But checking the noise first is likely less work

Reply:
But yeah, the corners of tiles should be at round numbers, and the centre should be at .5 and .5
In the upper right corner it tells you the current map coordinates of the mouse cursor if you wanna check

Reply:
It is for the obstacles. 
![image](https://user-images.githubusercontent.com/15359953/54624873-82090d00-4a6e-11e9-9cc1-27da2e33f1c2.png)
(Thats without moving the position)

Reply:
Oh, I know, you can get rid of the `- 0.5`s in the `draw_ellipse_outline(noise.location.x + 0.5, noise.location.y + 0.5, ...` calls. And probably also make the random noise coordinates real numbers instead of `int`s. (But that that last, since it's easier to check it works if they're still `int`s)

Reply:
Oh well... Fixed it